source/fe/mapping_c1.cc

```diff
@@ -149,11 +147,14 @@ MappingC1<2>::MappingC1Generic::add_line_support_points (const Triangulation<2>:
             };
         }
       else
-        // not at boundary
+        // not at boundary, so just use scaled Gauss-Lobatto points (i.e., use
+        // plain straight lines).
         {
-          static const StraightBoundary<dim> straight_boundary;
-          straight_boundary.get_intermediate_points_on_line (line, line_points);
-          a.insert (a.end(), line_points.begin(), line_points.end());
+          for (unsigned int point_n = 1; point_n < 3; ++point_n)
+            {
+              a.push_back((1.0 - quad_points.point(point_n)[0])*line->vertex(0) +
+                          (quad_points.point(point_n)[0]*line->vertex(1)));
```

Comment:
the loop is unnecessarily complicated here -- just repeat the `push_back` call twice for the appropriate points. maybe add a comment that says why the second and third point are the correct ones